README.md

```diff
@@ -292,4 +290,96 @@ class Example extends StatelessWidget {
     );
   }
 }
-```
\ No newline at end of file
+```
+
+### ReactiveWidget
+Widget built to encapsulate StreamBuilder.
+
+Similar to **ReactiveBuilder** but doesn't require `BuiltContext` in its parameters
+
+**stream** `Stream<T>` that controls the widget.
+
+**initialData** can be used to provide an initial value just as it does in StreamBuilder.
+
+**widget** lis a method that will be cal when the stream receive data.
+
+**placeHolderWidget** is a method that will be cal when the stream is initialized without data or when receive a null data. By default this method returns a CircularProgressIndicator.
+
+**errorWidget** is a method that will be call when the stream receive a error. By default this method returns a Text containing the error.
+
+#### ReactiveWidget Example
+``` dart
+class Animal {
+  String name;
+  int age;
+}
+
+class Example extends StatelessWidget {
+  final Stream<Animal> stream;
+
+  Example(this.stream);
+
+  @override
+  Widget build(BuildContext context) {
+    return ReactiveBuilder<Animal>(
+      stream: stream,
+      builder: (data) {
+        return ListTile(
+          title: Text(data.name),
+          subtitle: Text("${data.age}"),
+        );
+      },
+    );
+  }
+}
+```
+
+
+### RxCommandHandlerMixin 
+
+Adds ability to listen om RxCommand events inside Stateless and Stateful Widgets.
+
+Add **RxCommandHandlerMixin** to your `Stateless Widget` or  
+**RxCommandStatefulHandlerMixin**  to your `Stateful Widget` and override 
+**RxCommandListener get commandListener**.  
+ 
+It will add your widget to get an ability to listen for events from command and make actions 
+specified in **RxCommandListener** when its streams fire events.
+
+Mixin will take care of RxCommandListener init dispose. This way you can keep your widget stateless. 
+Or if it is already stateful - don't need to write code for that. 
+
+#### RxCommandHandlerMixin Example
+
+```dart
+class CommandWidget extends StatelessWidget with RxCommandHandlerMixin {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Center(
+        child: RaisedButton(
+          child: Text('Test Command'),
+          onPressed: _command,
+        ),
+      ),
+    );
+  }
+
+  @override
+  RxCommandListener get commandListener => RxCommandListener<void, DateTime>(

```

Comment:
Sorry I hadn't time to look at this earlier.
Why did you choose this way with the override instead to give the mixin a `registerCommandHandler()` that has all the parameters a listener has? This way you always can only listen to one Command ins a Widget.

Reply:
Posted reason for that above. It can be Iterable of listeners then. 

I've decided not to go with `registerStreamHandler()`. It has two disadvantages to my mind. 
First - you can add mixin to your class and forget calling `register..()`. With my approach its necessary getter field. 
Second - I'm not expecting to put anything executable in build() method. Even Flutter documentation states this: `This method can potentially be called in every frame and should not have any side effects beyond building a widget.`

Reply:
oh sorry I missed your comment. sure that is a valid point. although the thing with executing something in the build method isn't that critical. I will merge this now